pkg/util/aggregatingqueue/queue_test.go

```diff
@@ -14,55 +14,71 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func newMockFlush[T any]() (callback func([]T), getAccumulator func() [][]T) {
+func newMockFlush[T any]() (callback func([]T), waitCallbackCalled func(int), getAccumulator func() [][]T) {
 	accumulator := [][]T{}
-	var mutex sync.RWMutex
+	nbCalled := 0
+	var mutex sync.Mutex
+	cond := sync.NewCond(&mutex)
 
 	callback = func(elems []T) {
 		mutex.Lock()
 		defer mutex.Unlock()
 		accumulator = append(accumulator, elems)
+		nbCalled++
+		cond.Signal()
+	}
+
+	waitCallbackCalled = func(nb int) {
+		mutex.Lock()
+		defer mutex.Unlock()
+		for nbCalled < nb {
+			cond.Wait()
+		}
 	}
 
 	getAccumulator = func() [][]T {
-		mutex.RLock()
-		defer mutex.RUnlock()
+		mutex.Lock()
+		defer mutex.Unlock()
 		return accumulator
 	}
 
 	return
 }
 
 func TestQueue(t *testing.T) {
-	callback, accumulator := newMockFlush[int]()
+	callback, wait, accumulator := newMockFlush[int]()
 	cl := clock.NewMock()
 	queue := newQueue(3, 1*time.Minute, callback, cl)
 
 	for i := 0; i <= 10; i++ {
 		queue <- i
 	}
 
+	wait(3) // wait callback to have been called 3 times
+
 	assert.Equal(
```

Comment:
Is it possible that this assert fail because a 4th callback was called ?

Reply:
IMO it would be very unlikely

Reply:
No. Because the 4th call is triggered by [the expiration of a timer](https://github.com/DataDog/datadog-agent/blob/018bec69fd753d9fac543252245a3c29a5717270/pkg/util/aggregatingqueue/queue.go#L49) and the timer uses a mocked clock that doesn’t progress unless an explicit call to `cl.Add(…)` is done.